src/librustc/ty/mod.rs

```diff
@@ -1753,17 +1753,19 @@ bitflags! {
     pub struct AdtFlags: u32 {
         const NO_ADT_FLAGS        = 0;
         const IS_ENUM             = 1 << 0;
-        const IS_PHANTOM_DATA     = 1 << 1;
-        const IS_FUNDAMENTAL      = 1 << 2;
-        const IS_UNION            = 1 << 3;
-        const IS_BOX              = 1 << 4;
+        const IS_UNION            = 1 << 1;
+        const IS_STRUCT           = 1 << 2;
```

Comment:
`IS_STRUCT` == `!IS_ENUM && !IS_UNION`.
There's even a ready function for this - `AdtDef::is_struct` :)

Reply:
Definitely less readable though, hence my chance. No?

Reply:
Ok, readability is good, and the extra flag doesn't waste space or anything.

Reply:
@petrochenkov That's why I thought (it's a `u32` regardless). Thanks. :-)